ios/brave-ios/Sources/PlaylistUI/PlaylistSidebarList.swift

```diff
@@ -0,0 +1,325 @@
+              } label: {
+                Label("Open In New Private Tab", braveSystemImage: "leo.product.private-window")
+              }
+              ShareLink(item: url) {
+                Label("Share", braveSystemImage: "leo.share.macos")
+              }
+              Divider()
+            }
+            if folders.count > 1 {
+              Picker(
+                selection: Binding(
+                  get: { selectedFolderID },
+                  set: {
+                    PlaylistItem.moveItems(items: [item.objectID], to: $0)
+                  }
+                )
+              ) {
+                ForEach(folders, id: \.objectID) { folder in
+                  Label(folder.title ?? "", braveSystemImage: "leo.product.playlist")
+                    .tag(folder.id)
+                }
+              } label: {
+                Label("Move…", braveSystemImage: "leo.folder.exchange")
+              }
+              .pickerStyle(.menu)
+            }
+            Button(role: .destructive) {
+              PlaylistManager.shared.delete(item: .init(item: item))
+              if selectedItemID == item.id {
+                selectedItemID = nil
+              }
+            } label: {
+              Label("Delete", braveSystemImage: "leo.trash")
+            }
+          }
+        }
+      }
+    }
+    .frame(maxWidth: .infinity, alignment: .leading)
+    .onAppear {
+      for item in items {
+        guard let uuid = item.uuid else { continue }
+        downloadStates[uuid] = PlaylistManager.shared.state(for: uuid)
+      }
+    }
+    .onReceive(PlaylistManager.shared.downloadStateChanged) { output in
+      downloadStates[output.id] = output.state
+    }
+    .onReceive(PlaylistManager.shared.downloadProgressUpdated) { output in
+      downloadProgress[output.id] = output.percentComplete
+    }
+  }
+}
+
+struct PlaylistSidebarListHeader: View {
+  var folders: [PlaylistFolder]
+  @Binding var selectedFolder: PlaylistFolder
+  var selectedItemID: PlaylistItem.ID?
+  @Binding var isPlaying: Bool
+  @Binding var isNewPlaylistAlertPresented: Bool
+  @Binding var isEditModePresented: Bool
+
+  @State private var totalSizeOnDisk = Measurement<UnitInformationStorage>(value: 0, unit: .bytes)
+  @State private var isDeletePlaylistConfirmationActive: Bool = false
+  @State private var isRenamePlaylistAlertPresented: Bool = false
+
+  @MainActor private func calculateTotalSizeOnDisk(for folder: PlaylistFolder) async {
+    // Since we're consuming CoreData we need to make sure those accesses happen on main
+    let items = folder.playlistItems ?? []
+    var totalSize: Int = 0
+    for item in items {
+      guard let cachedData = item.cachedData else { continue }
+      totalSize += await Task.detached {
+        // No CoreData usage allowed in here
+        do {
+          var isStale: Bool = false
+          let url = try URL(resolvingBookmarkData: cachedData, bookmarkDataIsStale: &isStale)
```

Comment:
We can use the `cachedData` extension?

Reply:
Yeah, I added that extension shortcut near the end of this so missed replacing this